squash/dashboard/views.py

```diff
@@ -101,100 +103,84 @@ class DefaultsViewSet(DefaultsMixin, viewsets.ViewSet):
-    """API endpoint for listing AMx app data"""
-
-    def list(self, request):
-
-        # TODO: filter by job id
-        job = Job.objects.latest('pk')
-        blob_serializer = BlobSerializer(job)
-
-        # TODO: filter by metric
-        measurement = Measurement.objects.filter(job=job)[0]
-        measurement_serializer = MeasurementSerializer(measurement)
-
-        data = {}
-        metadata = {}
+        queryset = Metric.objects.values_list('metric', flat=True)
 
-        if measurement_serializer.data['metadata']:
-
-            metadata['metadata'] = \
-                eval(measurement_serializer.data['metadata'])
-
-            # datasets used in this measurement
-            blobs = metadata['metadata'].pop('blobs')
-
-            matched_dataset_id = blobs['matchedDataset']
-            astrom_model_id = blobs['astromModel']
-
-            # given the dataset ids, get the actual data
-            # from the Job model
-            data_blobs = eval(blob_serializer.data['blobs'])
-
-            for blob in data_blobs:
-                if blob['identifier'] == matched_dataset_id:
-                    data['matchedDataset'] = blob['data']
-                elif blob['identifier'] == astrom_model_id:
-                    data['astromModel'] = blob['data']
-
-        return response.Response({**data, **metadata}) # noqa
+        if 'AM1' in queryset:
+            metric = 'AM1'
+        else:
+            metric = queryset.latest('pk')
 
+        snr_cut = '100'
+        window = 'months'
 
-class PAxViewSet(DefaultsMixin, CacheResponseMixin, viewsets.ViewSet):
-    """API endpoint for listing PAx app data"""
+        return {'ci_id': ci_id, 'ci_dataset': ci_dataset,
+                'metric': metric, 'snr_cut': snr_cut,
+                'window': window}
 
     def list(self, request):
+        defaults = self.get_defaults()
+        return response.Response(defaults)
+
 
-        # TODO: filter by job id
-        job = Job.objects.latest('pk')
-        blob_serializer = BlobSerializer(job)
+class BokehAppViewSet(DefaultsMixin, viewsets.ViewSet):
 
-        # TODO: filter by metric
-        measurement = Measurement.objects.filter(job=job)[0]
-        measurement_serializer = MeasurementSerializer(measurement)
+    def get_app_data(self, ci_id, ci_dataset, metric):
 
         data = {}
-        metadata = {}
 
-        if measurement_serializer.data['metadata']:
+        blobs = Job.objects.filter(ci_id=ci_id,
+                                   ci_dataset=ci_dataset).values('blobs')
+
+        metadata = Measurement.\
+            objects.filter(metric=metric, job__ci_id=ci_id,
+                           job__ci_dataset=ci_dataset).values('metadata')
+
+        if metadata.exists():
+            # workaround for getting item from queryset
+            metadata = metadata[0]['metadata']
+            if metadata:
+                metadata = literal_eval(literal_eval(metadata))
```

Comment:
The metadata is quoted, so the nested literal_eval is to remove the quoted string.